frontend/src/pages/SearchPage.tsx

```diff
@@ -10,20 +11,34 @@ import { useParams } from 'react-router-dom';
 export type SearchState = {
   searchQuery: string;
   page: number;
+  selectedCuisines: string[];
 };
 
 const SearchPage = () => {
   const { city } = useParams<{ city: string }>();
   const [searchState, setSearchState] = useState<SearchState>({
     searchQuery: '',
     page: 1,
+    selectedCuisines: [],
   });
+
+  const [isExpanded, setIsExpanded] = useState<boolean>(false);
+
   const { results, isLoading } = useSearchRestaurants(searchState, city);
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/44#issuecomment-2410860532>#3a7709</a><span> - 10/14/2024, 11:08 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The useSearchRestaurants hook is called on every render without any memoization, which may lead to unnecessary API calls.
</div>
<div id="fix">
<b>Fix:</b> Use the useMemo hook to memoize the dependencies of useSearchRestaurants and prevent unnecessary API calls.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -25,7 +25,11 @@
    const [isExpanded, setIsExpanded] = useState<boolean>(false);
 
 -  const { results, isLoading } = useSearchRestaurants(searchState, city);
 +  const searchDependencies = useMemo(() => ({
 +    ...searchState,
 +    city
 +  }), [searchState, city]);
 +  const { results, isLoading } = useSearchRestaurants(searchDependencies);
 
    const setSelectedCuisines = (cuisines: string[]) => {
      setSearchState((prevState) => ({
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged